doc/doc_examples_to_gallery.py

```diff
@@ -12,42 +12,58 @@
 
 """
 import os
-
+import time
 basedir = os.getcwd()
 
-examples_dir = os.path.join(os.getcwd(), '../examples')
+examples_dir = os.path.abspath(os.path.join(basedir, '..', 'examples'))
 files = [fn for fn in os.listdir(examples_dir) if fn.startswith('doc_')]
 
 examples_documentation_dir = os.path.join(examples_dir, 'documentation')
 os.makedirs(examples_documentation_dir, exist_ok=True)
 
+
+scripts_to_run = []
+
 with open(os.path.join(examples_documentation_dir, 'README.txt'), 'w') as out:
     out.write("Examples from the documentation\n")
     out.write("===============================\n\n")
     out.write("Below are all the examples that are part of the lmfit documentation.")
 
 for fn in files:
+    inp_path = os.path.join(examples_dir, fn)
+    with open(inp_path, 'r') as inp:
+        script_text = inp.read()
+
     gallery_file = os.path.join(examples_documentation_dir, fn[4:])
     with open(gallery_file, 'w') as out:
-        if fn == 'doc_model_loadmodel.py':
-            msg = ('This example *does* actually work, but for some reason the '
-                   'conversion using sphinx-gallery fails....')
-            out.write('"""\n{}\n{}\n\n{}\n\n"""\n'.format(fn, "="*len(fn), msg))
-        else:
-            out.write('"""\n{}\n{}\n\n"""\n'.format(fn, "="*len(fn)))
-    os.system('cat {} >> {}'.format(os.path.join(examples_dir, fn), gallery_file))
+        msg = ""
+        if 'model_loadmodel.py' in fn:
+            msg = ('.. note:: This example *does* actually work, but running from within '
+                   ' sphinx-gallery fails to find symbols saved in the save file.')
+        out.write('"""\n{}\n{}\n\n{}\n"""\n'.format(fn, "="*len(fn), msg))
+        out.write('##\nimport warnings\nwarnings.filterwarnings("ignore")\n##\n')
```

Comment:
Yes, I was seeing 'matplotlib only using Agg backend' warnings showing up in the rendered results and it seemed distracting.   I think adding that "suppress warnings" would not be a great idea for the actual example code, but only when rendering the docs.  In fact, I tried making the `filterwarnings()` call only ignore `Matplotlib` warnings, but that did not work.  I wasn't sure why that was (maybe also because inside some `exec()` ??), so I just let it ignore all warnings.